### PR TITLE
[Shop] Sorting products by price

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | no |
| Related tickets |  |
| License         | MIT |

Together with @Zales0123 ;)

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
We should add a scenario about products with multiple variants and make sure we sort by the price of lowest variant.

File: src/Sylius/Bundle/ShopBundle/Resources/views/Product/Index/_main.html.twig [hunk 1]
```diff
@@ -8,7 +8,7 @@
     <div class="ui three column stackable grid" id="products">
         {% for product in resources.data %}
             <div class="column">
-                {% include '@SyliusShop/Product/_box.html.twig' %}
+                {% include '@SyliusShop/Product/_box.html.twig' with { 'product': product[0] } %}
```
Comment (michalmarcinkowski) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Product/Index/_main.html.twig]: Do we need this?

Comment (TheMadeleine) [inline src/Sylius/Bundle/ShopBundle/Resources/views/Product/Index/_main.html.twig]: @Zales0123 do we need this? 😆

**Comment by pjedrzejewski** [comment]
Thank you Magda and Mateusz! Nice work! :)

**Comment by TheMadeleine** [comment]
Hi @inspiran It may fail unfortunately :( This query does not use any aggregating function. I didn't have this issue locally since I'm using mysql 5.6.26.

@michalmarcinkowski @pjedrzejewski What do you suggest doing?

**Comment by marcoslibre** [comment]
This is a SQL standard. Omitting unaggregated columns leads to indeterminate behavior.
MySQL (prior 5.7) sloppy implementation of this is more like a bug than a feature.

**Pull request merged**